src/phoenix/server/api/mutations/chat_mutations.py

```diff
@@ -128,14 +137,38 @@ async def chat_completion(
                 if isinstance(chunk, TextChunk):
                     text_content += chunk.content
                 elif isinstance(chunk, ToolCallChunk):
-                    tool_call = ChatCompletionToolCall(
-                        id=chunk.id,
-                        function=ChatCompletionFunctionCall(
-                            name=chunk.function.name,
-                            arguments=chunk.function.arguments,
-                        ),
-                    )
-                    tool_calls.append(tool_call)
+                    if len(tool_calls) == 0:
+                        tool_call = ChatCompletionToolCall(
+                            id=chunk.id,
+                            function=ChatCompletionFunctionCall(
+                                name=chunk.function.name,
+                                arguments=chunk.function.arguments,
+                            ),
+                        )
+                        tool_calls.append(tool_call)
+                    else:
+                        tool_exists = False
+                        for i, tool_call in enumerate(tool_calls):
+                            if tool_call.id == chunk.id:
+                                tool_calls[i] = ChatCompletionToolCall(
+                                    id=tool_call.id,
+                                    function=ChatCompletionFunctionCall(
+                                        name=tool_call.function.name,
+                                        arguments=tool_call.function.arguments
+                                        + chunk.function.arguments,
+                                    ),
+                                )
+                                tool_exists = True
+                                break
+                        if not tool_exists:
+                            tool_call = ChatCompletionToolCall(
+                                id=chunk.id,
+                                function=ChatCompletionFunctionCall(
+                                    name=chunk.function.name,
+                                    arguments=chunk.function.arguments,
+                                ),
+                            )
+                            tool_calls.append(tool_call)
```

Comment:
Sort of feels like we need `tool_calls` to be a map of tool call id to value here.

Reply:
this is for adding to a span I think, in that case i think we should add it like we do other tool calls which is a list of tool calls not a map, fixed a bug before where we were setting tools as maps here https://github.com/Arize-ai/phoenix/pull/5203

Reply:
okay, yes we can make dict and call .values

Reply:
much better